src/main/kotlin/org/rust/lang/core/psi/RsFile.kt

```diff
@@ -79,11 +80,17 @@ class RsFile(
     val isIncludedByIncludeMacro: Boolean get() = true
 //    val isIncludedByIncludeMacro: Boolean get() = cachedData.isIncludedByIncludeMacro
 
+    /** Used for in-memory macro expansions */
+    @Volatile
+    private var forcedCachedData: CachedData? = null
+
     private val cachedData: CachedData
         get() {
             val originalFile = originalFile
             if (originalFile != this) return (originalFile as? RsFile)?.cachedData ?: EMPTY_CACHED_DATA
 
+            forcedCachedData?.let { return it }
```

Comment:
Is it ok that we use `forcedCachedData` without any modification tracker check?

Is it possible scenario like this:
* We inherit cached data from some file
* Cached data of that file was changed but we still have old `forcedCachedData` here

Reply:
It's OK in the case of in-memory macro expansion, I've added a comment about it. But in other cases I can't be sure, so I switched to using a lazy CachedData in these cases